### PR TITLE
Remove Model & Formats docs from main repo

### PR DESCRIPTION
**Pull request by hflynn**
These docs are being moved to the new OME-Model repo see https://github.com/ome/ome-model/pull/19 and the merge job has already been updated to build from there so this PR removes the duplicated content from this repo and updates the README and CONTRIBUTING to reflect this.

I'll open a separate PR to update the contributing developer docs.

### CODE REVIEW

**Comment by hflynn** [comment]
This should have no impact on the docs jobs on Jenkins, you just need to review the first commit and check everything is green.

**Comment by rleigh-codelibre** [comment]
```
% git grep formats.release
Makefile:ANT_FORMATS_RELEASE := -Dformats.release="$(FORMATS_RELEASE)"
common/rules.xml:  <property name="formats.release" value="${omero.shortversion}"/>
common/rules.xml:        <env key="FORMATS_RELEASE" value="${formats.release}"/>
common/rules.xml:        <env key="FORMATS_RELEASE" value="${formats.release}"/>
common/rules.xml:        <env key="FORMATS_RELEASE" value="${formats.release}"/>
contributing/ci-docs.txt:               #. |ssh-doc| :file:`formats-release.tmp`
```

These can probably all be removed as well.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 This repository hosts the documentation for OMERO, and **on the** `develop`
-**branch only**, the OME Data Model and File Formats documentation and the
-guidance for contributing developers. The Bio-Formats documentation is part of
-the
-[main Bio-Formats repository](https://github.com/openmicroscopy/bioformats).
+**branch only**, the guidance for contributing developers. The Bio-Formats
+documentation is part of the
+[main Bio-Formats repository](https://github.com/openmicroscopy/bioformats)
+and similarly, other OME projects have their documentation within the same
+repository as the code.
```
Comment (rleigh-codelibre) [inline CONTRIBUTING.md]: the code they are documenting?  For a bit of additional clarity.

**Comment by rleigh-codelibre** [comment]
Everything here looks good to merge.  Just a few minor points about additional bits which might be deleted at the same time.

**Comment by hflynn** [comment]
I'll push another commit to clean up those other bits thanks

**Comment by hflynn** [comment]
This should be all sorted now @rleigh-codelibre the only thing I've left is the reference within the contributing folder which I will fix on my other PR #1595

**Comment by rleigh-codelibre** [comment]
This is working fine, and is good to merge.

**Pull request merged**